### ISSUE TITLE
Support for Selenium 4

### ISSUE THREAD

**Issue by lawcrens** [microsoft/WinAppDriver#1839]
Hello,

Wanted to request that WinAppDriver is supported on Selenium 4. Selenium 3 has been deprecated for a year

**Comment by pm90pl** [comment]
What do you exactly mean by 'Selenium 4 support '? WinAppDriver implements W3 WebDriver standard. Selenium 4 is a client implementation that consumes W3 WebDriver API.

**Comment by DLightstone** [comment]
One may stand to gain by considering WinAppDriver to be a moot which serves to protect Microsoft to firms that would like to nibble at their market positions (ie a product entrance barrier)

A moot does not have to be supported and isn't.

I would consider converting the application into something that can be tested without using WinAppDriver

**Comment by PavanMudigondaTR** [comment]
I have the same question ! Please someone from Microsoft please answer !

**Comment by farnsword** [comment]
Hey everyone.
Can anyone advise if it is going to happen?
It is a kind of blocker.

**Comment by pm90pl** [comment]
Yeah. So, the current version is not compatible with W3C WebDriver standard what makes it incompatible with Selenium 4. There is no clear roadmap for WAD and it's not open source so you have to deal it on your own. Sorry.

**Comment by farnsword** [comment]
Well, at this point I only need to update the selenium version to 4.0 as using the latest Chrome version requires Selenium v4, so I stuck with Chrome v95.
As a last resort - this repository is under an MIT license which allows forking and modification, so may try to build our own version of it.
Thanks for the answer.

**Comment by pm90pl** [comment]
But this repo contains only examples, not WinAppDriver implementation.

**Comment by farnsword** [comment]
Oh my...thanks for pointing that out.
Tough luck then.

**Comment by PavanMudigondaTR** [comment]
I have this email address of Microsoft team winappdriverteam@microsoft.com

**Comment by d2a-raudenaerde** [comment]
Has anybody sent them an email yet?

**Comment by DLightstone** [comment]
The dear folks at Microsoft read the postings.

To determine it for yourself, post a disparaging comment and wait for it to be hidden or deleted

No reply means minimal support or no support

**Comment by PavanMudigondaTR** [comment]
Sent. I request everyone to send a email.  To save time you can copy and send the below.

[winappdriverteam@microsoft.com](mailto:winappdriverteam@microsoft.com)
----------------------------------------

Hi WinAppDriver Team at Microsoft,
 
We are looking for support for Selenium 4 Java for WinAppDriver.  We would appreciate if someone could look at this thread and respond back on your plans !
 
 
https://github.com/microsoft/WinAppDriver/issues/1839#issuecomment-1382019123a

**Comment by d2a-raudenaerde** [comment]
@PavanMudigondaTR thanks for the template!. Would you please edit it and add 'Thanks for all the great work so far!' as first line? Thanks :)

**Comment by nguyenquangtay** [comment]
sent also, thanks! :)

**Comment by dmridul1827** [comment]
thanks everyone for bringing it up. I am also facing same issue while migrating winappdriver to work with Sel 4.

Any further update or resolution, anyone has? Would be grateful!

**Comment by ShinjiruPP** [comment]
Are there some updates??? About Selenium4 + WinAppDriver ??

**Comment by fastinggirl** [comment]
Any update on this?

**Comment by sunrise-soft** [comment]
Hi there!
Seem this is an abandoned project :(
I just wonder does Microsoft applies test automation for their Desktop applications and if yes, which one they're using

**Comment by RamaNarasimhaChanda** [comment]
Hi All, Since we are not having update from micro soft on this . What is the alternative approach other's are using can you please share.

**Comment by daluu** [comment]
Well, obviously, for this issue, I can think of 2 workarounds/alternatives:

- don't use Selenium 4 while using WinAppDriver, until a fix is available if ever (so stay on Selenium 3 or 2, etc.). For this particular issue, if you need to use Selenium 4, maybe use dual approach and dual version dependency management. Use Selenium 4 for web/mobile automation, then use Selenium 3 for now for desktop automation, within your test framework.

- look into WinAppDriver alternative tools (there are some), and see which of those support (1) WebDriver API/protocol, and (2) Selenium 4 version of the protocol.

There are alternative tools mentioned in various comments in #1550, and I sort of compiled a list in https://github.com/Accruent/robotframework-applicationlibrary/issues/72#issuecomment-1745389341

**Comment by AG-QALab** [comment]
- Install Nodejs and WinAppDriver
- Install Appium 2+ version : npm i -g appium
- Install appium windows driver : appium driver install --source=npm appium-windows-driver
- run appium command
- execute the scripts

Its working.

**Comment by hritha** [comment]
@AG-QALab - can u please let me know if this works with Java winAppdriver scripts too

> * Install Nodejs and WinAppDriver
> * Install Appium 2+ version : npm i -g appium
> * Install appium windows driver : appium driver install --source=npm appium-windows-driver
> * run appium command
> * execute the scripts
> 
> Its working.

**Comment by alatriste1982** [comment]
Any update from MS on this one?
